CMakePresets.json

```diff
@@ -0,0 +1,52 @@
+{
+    "version": 6,
+    "cmakeMinimumRequired": {
```

Comment:
I had to go update my cmake to test build the project. We shouldn't be bound by, for example, the version shipped in Ubuntu 20.04, but do we want to require current `cmake` as of when the commit was done? 
Not a major problem, but if we want people to use current dev tools we might want to link to their instructions on where to get them. 
There's a good chance we'll be requiring unreleased C++ compilers, in any case. So an advanced set of tools is something people will need to figure out how to install.

Reply:
We'll want CI for these builds. Given that, I think there's a pretty clear implicit requirement that we have a lowest common denominator version of each build tool, including CMake. And I believe using the CMake naturally provided in those environments (like a cmake package) is appropriate compared to downloading binaries or building cmake from source.

As far as Ubuntu goes, the new 24.04 LTS should have a new enough CMake for the Make lists.txt in this project.